docs/structured-streaming-programming-guide.md

```diff
@@ -978,11 +978,11 @@ val windowedCounts = words
 Dataset<Row> words = ... // streaming DataFrame of schema { timestamp: Timestamp, word: String }
 
 // Group the data by window and word and compute the count of each group
-Dataset<Row> windowedCounts = words
-    .withWatermark("timestamp", "10 minutes")
+Dataset<Row> wordsWatermark = words.withWatermark("timestamp", "10 minutes");
+Dataset<Row> windowedCounts = wordsWatermark
     .groupBy(
-        functions.window(words.col("timestamp"), "10 minutes", "5 minutes"),
-        words.col("word"))
+        functions.window(wordsWatermark.col("timestamp"), "10 minutes", "5 minutes"),
```

Comment:
I guess `functions.col("timestamp")` is exactly equivalent to the `$"timestamp"` in Scala code example. Same applies to col("word").

Actually Java code example seems to be written a bit verbosely. `import static org.apache.spark.sql.functions.*;` would shorten the code and remove `functions.` all the places - actually Scala code example assume that the import is placed before.

Reply:
@HeartSaVioR   Thank you for your help.